### ISSUE TITLE
Feature: Client auto updater

### ISSUE THREAD

**Issue by Mhowser** [zeroeightysix/KAMI#83]
It would be pretty great if KAMI could auto-update itself when a new release is available, instead of downloading the jar manually

**Comment by ChristoferMinestar** [comment]
Add an option to select what branch

**Comment by ChristoferMinestar** [comment]
Also, if this simply updated the updated files and added the new files, instead of replacing it completely would make if more efficient and if you've modded the files like to disable the injection it keeps those modifications

**Comment by wine** [comment]
I mean, I already have auto-build setup on my fork, currently builds and releases the artifact if the commit is marked as a release.
Wouldn't be hard to add to the official KAMI repo either

**Comment by dominikaaaa** [comment]
Just leaving this here for reference, this is a pain in the ass to support on Windows due to Windows FS locking the mod file, even though it's technically loaded into ram, the only way to do it is with the game closed or by replacing the file's classes yourself, which screws any idea of adding version numbers in file names.

**Comment by Mhowser** [comment]
What about us Linux users ;)

**Comment by dominikaaaa** [comment]
It is *much* easier on Linux. It seems like it's almost impossible on OSX, I couldn't get Java's File.delete() to work, I probably had to ask for permission or something.

On Linux you can just delete the old file and download the new one no problem, without any issues.

**Comment by JerryWester** [comment]
Proof that Linux is superior

**Comment by RealIndrit** [comment]
DONT FKN HOT RELOAD, ezz fix...

**Comment by Mhowser** [comment]
> Proof that Linux is superior

Always has been

**Comment by LordMZTE** [comment]
I don't think this is a good idea. kami is very easy to update due to being a fabric mod (unlike other mods like impact). This in my opinion may also not be desirable. For example what if I work on my fork and it thinks it's an outdated version, or I wanna use an old version for whatever reason. That could get really annoying.

**Comment by 5HT2** [comment]
KAMI isn't updated often enough to warrant this either

**Comment by ghost** [comment]
> It is _much_ easier on Linux. It seems like it's almost impossible on OSX, I couldn't get Java's File.delete() to work, I probably had to ask for permission or something.
> 
> On Linux you can just delete the old file and download the new one while running, no problem, without any issues.

Even though it seems like KAMI isn't getting an auto-updater one approach you could take is having a daemon type process that handles updating.

**Comment by Mhowser** [comment]
This has already been done with Wurst client.
(GPL v3) https://github.com/Wurst-Imperium/Wurst7/blob/master/src/main/java/net/wurstclient/update/WurstUpdater.java

I think you guys could use this as well.

**Comment by 5HT2** [comment]
That doesn't do anything except tell you to download the latest version lol

**Comment by Mhowser** [comment]
Sorry I dont really understand code. Is this the right one? (GPL v3) https://github.com/Wurst-Imperium/Wurst7/blob/master/src/main/java/net/wurstclient/update/Version.java

**Comment by 5HT2** [comment]
No, that just checks if the current version is the latest one, and gives you a link to the Wurst website

**Comment by Mhowser** [comment]
I don't understand code unfortunately, sorry, check out version.java in the
update folder.

**Comment by 5HT2** [comment]
That is the same Version.java in the update folder, it only checks if the current version is out of date lol

**Comment by Mhowser** [comment]
That was an email that sent too late sorry.
